froact.lua

```diff
@@ -171,13 +171,81 @@ local function newC<Hooks>(roact: any, hooks: HookFunction<any, Hooks>, unpureBy
 	end
 end
 
+type Ref = { current: any }
+local function newCreateRef(roact: any): () -> Ref
+	return roact.createRef
+end
+
+type Binding<T> = { getValue: (self: Binding<T>) -> T }
+local function newCreateBinding(roact: any): <T>(default: T) -> (Binding<T>, (T) -> ())
+	return roact.createBinding
+end
+
+type BindingPairs<T...> = { map: <O>(self: BindingPairs<T...>, f: (T...) -> O) -> Binding<O> }
+local function newJoin(roact: any): (
+	(<A>(b1: Binding<A>) -> BindingPairs<A>)
+	& (<A, B>(b1: Binding<A>, b2: Binding<B>) -> BindingPairs<A, B>)
+	& (<A, B, C>(b1: Binding<A>, b2: Binding<B>, b3: Binding<C>) -> BindingPairs<A, B, C>)
+	& (<A, B, C, D>(
+		b1: Binding<A>,
+		b2: Binding<B>,
+		b3: Binding<C>,
+		b4: Binding<D>
+	) -> BindingPairs<A, B, C, D>)
```

Comment:
I should probably explain this atrocity

Reply:
based off https://github.com/fewkz/rx-luau/blob/49adba616badfba87cbe98083bb74ba098052a73/src/pipe.luau#L6-L16